rules/falco_rules.yaml

```diff
@@ -3420,3 +3420,35 @@
 # Application rules have moved to application_rules.yaml. Please look
 # there if you want to enable them by adding to
 # falco_rules.local.yaml.
+
+
+- macro: rocketmq_check_netty_server
+  condition: (proc.name contains NettyServer and proc.env contains ROCKETMQ)
+
+- macro: rocketmq_check_payload_for_conf_update
+  condition: ((fd.typechar='4' or fd.typechar='6') and evt.type=read and evt.buffer contains 'code":25' and evt.buffer contains serializeTypeCurrentRPC and evt.buffer contains JSON)
+
+- rule: Apache Rocketmq CVE-2023-33246 Exploit Detection
+  desc: >
+    CVE-2023-33246 is exploited at the function used to update configuration & can be detected by looking at the buffer for broker code-25 and serializeTypeCurrentRPC function.
+  condition: >
+    rocketmq_check_netty_server and rocketmq_check_payload_for_conf_update and evt.dir=<
+  output: exploit detected CVE-2023-33246 (event=%evt.type server_ip=%fd.sip server_port=%fd.sport proto=%fd.l4proto parent=%proc.pname process=%proc.name command=%proc.cmdline container_id=%container.id)
+  priority: CRITICAL
+  tags: [host, container, CVE_2023_33246, rocketmq, apache,Mitre_Initial_Access, Mitre_Execution,TA0001, TA0002, T1133, T1059]
+
+
+- macro: rocketmq_check_env_variable
+  condition: (proc.env contains ROCKETMQ)
+
+- macro: rocketmq_check_properties_file_write_conf
+  condition: (fd.nameraw contains conf and fd.nameraw endswith '.properties')
+
+- rule: Apache Rocketmq CVE-2023-33246 Exploit Attempt
+  desc: >
+    Look for the file ends with '.properties' created under folder 'CONF' upon exploit attempt
+  condition: >
+    rocketmq_check_env_variable and open_write and rocketmq_check_properties_file_write_conf and evt.dir=<
```

Comment:
Once we merge the rules adoption and maturity framework proposal I will work on creating an official style guide and more docs which will include expected review timelines and testing procedures and more in order to set clearer expectations as we all are spread across multiple repos ... just FYI, also I still need to become official approver here first 🙃 

```
open_write and evt.dir=< and rocketmq_check_env_variable and rocketmq_check_properties_file_write_conf 
```

Reply:
Sure, a style guide definitely helps in the matter of ease of use and readability. In fact, naming conventions when we develop detections with various products are needed in my opinion. Here I used "product_action" as a convention but I am happy to take suggestions from the community. 

Also, I would suggest having a field or tag to know whether the rule detects an anomaly or active exploit/intrusion. I would prefer using tags `anomaly`, `dfir`, `exploit`. 

`Anomaly` - indicates there's an anomaly. (Default falco_rules triggers many anomalies and upon investigation, it results out as false positives)

`DFIR` - indicates there's actually an incident that occurred (which are certain based on intelligence or knowledge base), a few examples cryptominers, ransomware, webshells etc..

`Exploit` - Indicates there's an exploit activity observed (mostly specific to CVE/known exploits)

Given these tags helps me to run only the ones which I needed most based on my exercise. Again, we can work on the appropriate naming conventions. This is just a suggestion coming from experience leading Detection and Engineering team.

Reply:
Sure, a style guide definitely helps in the matter of ease of use and readability. In fact, naming conventions when we develop detections with various products are needed in my opinion. Here I used "product_action" as a convention but I am happy to take suggestions from the community. 

Also, I would suggest having a field or tag to know whether the rule detects an anomaly or active exploit/intrusion. I would prefer using tags `anomaly`, `dfir`, `exploit`. 

`Anomaly` - indicates there's an anomaly. (Default falco_rules triggers many anomalies and upon investigation, it results out as false positives)

`DFIR` - indicates there's actually an incident that occurred (which are certain based on intelligence or knowledge base), a few examples cryptominers, ransomware, webshells etc..

`Exploit` - Indicates there's an exploit activity observed (mostly specific to CVE/known exploits)

Given these tags helps me to run only the ones which I needed most based on my exercise. Again, we can work on the appropriate naming conventions. This is just a suggestion coming from experience leading Detection and Engineering team.